aptos-move/aptos-aggregator/src/aggregator_extension.rs

```diff
@@ -465,22 +462,19 @@ impl AggregatorData {
         }
     }
 
-    pub fn snapshot(&mut self, id: AggregatorID) -> PartialVMResult<AggregatorID> {
+    pub fn snapshot(&mut self, id: AggregatorID, max_value: u128) -> PartialVMResult<AggregatorID> {
```

Comment:
Why do we need to have max_value as extra argument? We can obtain the max_value based on the aggregator id input right?

Reply:
actually, we need it in the case aggregator was not touched in this transaction at all. (to in the end be able to create AggregatorChange for it - which has max_value).

we could set it to max_value in that case, but then it would be inconsistent in the MVHashMap, which I don't like

Reply:
Let's call it aggregator_max_value (or smt that highlights this max value is guaranteed to correspond to the aggregator ID)? was also confusing for me.